app/controllers/communities_controller.rb

```diff
@@ -1,70 +1,98 @@
 class CommunitiesController < ApplicationController
-  before_action :authenticate_account!, except:  [ :index]
-  before_action :set_community, only: [:show, :edit, :update, :destroy, :mod, :check_if_banned]
+  before_action :authenticate_account!, except: [:index]
+  before_action :set_community, only: %i[show edit update destroy mod]
   before_action :community_list
   before_action :check_if_banned, only: [:show]
   after_action :count_post_for_this_week, only: [:index]
 
   def index
     count_post_for_this_week
     @categories = Community::CATEGORIES
-    if(params.has_key?(:category))
-      @communities = Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
-    else
-      @communities = Community.order(created_at: :desc).page(params[:page]).per 7
-    end
-    @random_category_communities = Community.where(category: @categories.sample).order("post_count_this_week desc").limit(5)
+    @communities = if params.has_key?(:category)
+                     Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
+                   else
+                     Community.order(created_at: :desc).page(params[:page]).per 7
+                   end
```

Comment:
there are some alignment issues here? why is this happening? @tusharravindran

Reply:
This is due to refactoring code using the RuboCop gem .I will resolve it now